src/main/java/org/broadinstitute/hellbender/tools/pon/coverage/pca/HDF5PCACoveragePoNCreationUtils.java

```diff
@@ -531,178 +430,173 @@ static ReadCountCollection removeTargetsWithTooManyZeros(final ReadCountCollecti
-     * @param readCounts the input read counts.
-     * @return never {@code null}, with as many elements as targets in {@code readCounts}.
+     * @param readCounts the input and output read-count matrix.
      */
-    private static double[] calculateTargetFactors(final ReadCountCollection readCounts) {
-        return MatrixSummaryUtils.getRowMedians(readCounts.counts());
-    }
+    @VisibleForTesting
+    static void truncateExtremeCounts(final ReadCountCollection readCounts, final double percentile, final Logger logger) {
 
-    /**
-     * Calculates the absolute number of zeroes tolerated given the user requested percentage
-     * and the number of counts involved.
-     * @param totalCounts number of counts to check.
-     * @param percentage a value from 0 to 100, otherwise a exception will be thrown.
-     * @return a value between 0 and {@code totalCounts}.
-     */
-    private static int calculateMaximumZerosCount(final int totalCounts, final double percentage) {
-        return (int) Math.ceil(totalCounts * percentage / 100.0);
-    }
+        final RealMatrix counts = readCounts.counts();
+        final int targetCount = counts.getRowDimension();
+        final int columnCount = counts.getColumnDimension();
 
-    /**
-     * Determine the variance for each target in the PoN (panel targets).
-     *
-     * @param pon This pon must have the normalizedCounts populated.  Never {@code null}
-     * @return array of doubles where each double corresponds to a target in the PoN (panel targets)
-     */
-    private static double[] createTargetVariance(final PoN pon, final JavaSparkContext ctx) {
-        final TangentNormalizationResult allNormals = TangentNormalizer.tangentNormalizeNormalsInPoN(pon, ctx);
-        final RealMatrix allSampleProjectedTargets = allNormals.getTangentNormalized().counts();
+        // Create a row major array of the counts.
+        final double[] values = Doubles.concat(counts.getData());
 
-        return calculateRowVariances(allSampleProjectedTargets);
-    }
+        final Percentile bottomPercentileEvaluator = new Percentile(percentile);
+        final Percentile topPercentileEvaluator = new Percentile(100.0 - percentile);
+        final double bottomPercentileThreshold = bottomPercentileEvaluator.evaluate(values);
+        final double topPercentileThreshold = topPercentileEvaluator.evaluate(values);
+        long totalCounts = 0;
+        long bottomTruncatedCounts = 0;
+        long topTruncatedCounts = 0;
 
-    /**
-     * Calculate the variance for each row in a matrix.
-     *
-     * @param m Not {@code null}
-     * @return Not {@code null}.  double array with length {@code m.getRowDimension()}
-     */
-    static double[] calculateRowVariances(final RealMatrix m) {
-        Utils.nonNull(m, "Cannot calculate variances for a null matrix.");
-        return MatrixSummaryUtils.getRowVariances(m);
+        for (int i = 0; i < targetCount; i++) {
```

Comment:
perhaps, but i think this is beyond the scope of this PR.  @mbabadi might look into adding it to #634, but perhaps the same could be said for that PR as well.